ui/packages/catalog-ui-search/src/main/webapp/component/list-interactions/list-interactions.view.js

```diff
@@ -72,9 +77,40 @@ module.exports = Marionette.ItemView.extend({
     const newList = new this.model.constructor(copyAttributes)
     this.model.collection.add(newList)
   },
-  triggerAction(event) {
+  async triggerAction(event) {
     const url = event.currentTarget.getAttribute('data-url')
-    window.open(url)
+    try {
+      let res = await fetch(url, { Accept: 'application/json' })
+      if (!res.ok) {
+        throw Error(res.statusText)
+      }
+
+      if (res.headers.get('content-disposition') != null) {
+        const data = await res.text()
```

Comment:
Is it ok to use `text()` even if the attached file is a binary file?

Reply:
This is how we do it in `table-export.tsx` for all of those downloads that call `saveFile`.